pkg/basereconciler/reconcile_resources.go

```diff
@@ -212,31 +273,35 @@ func (r *Reconciler) GetDeploymentReplicas(ctx context.Context, d Deployment) (*
 // all controllers
 func (r *Reconciler) ReconcileOwnedResources(ctx context.Context, owner client.Object, crs ControlledResources) error {
 	// Calculate resources to enforce
-	resources := []LockedResource{}
+	resources := []basereconciler_types.LockedResource{}
 
 	for _, dep := range crs.Deployments {
 
-		currentReplicas, err := r.GetDeploymentReplicas(ctx, dep)
-		if err != nil {
-			return err
+		if dep.HasHPA {
+			currentReplicas, err := r.GetDeploymentReplicas(ctx, dep)
+			if err != nil {
+				return err
+			}
+			resources = append(resources,
+				basereconciler_types.LockedResource{
+					GeneratorFn:  r.DeploymentWithRolloutTriggers(dep.Template, dep.RolloutTriggers, currentReplicas),
+					ExcludePaths: append(DeploymentExcludedPaths, "/spec/replicas"),
+				})
+
+		} else {
+			resources = append(resources,
+				basereconciler_types.LockedResource{
+					GeneratorFn:  r.DeploymentWithRolloutTriggers(dep.Template, dep.RolloutTriggers, nil),
+					ExcludePaths: DeploymentExcludedPaths,
+				})
 		}
 
-		resources = append(resources,
-			LockedResource{
-				GeneratorFn: r.DeploymentWithRolloutTriggers(dep.Template, dep.RolloutTriggers, currentReplicas),
-				ExcludePaths: func() []string {
-					if dep.HasHPA {
-						return append(DeploymentExcludedPaths, "/spec/replicas")
-					}
-					return DeploymentExcludedPaths
-				}(),
-			})
```

Comment:
This was a bug where the number of replicas was not reconciled when HPA was disabled.